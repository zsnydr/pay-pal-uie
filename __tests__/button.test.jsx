import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../public/app/button';

describe('Button (Snapshot)', () => {
  it('Button renders a native button with given props', () => {
    const component = renderer.create(
      <Button name="testButton" />
    );
    const button = component.toJSON();
    expect(button).toMatchSnapshot();
  });
});

test('Button calls clickHandler on click', () => {
  const clickHandler = jest.fn();
  const buttonComp = mount(
    <Button name="testButton" clickHandler={clickHandler} />
  );
  const button = buttonComp.find('button');
  // expect(button.name).toEqual('testButton');
  button.simulate('click');
  expect(buttonComp.props().clickHandler).toBeCalled();
});
