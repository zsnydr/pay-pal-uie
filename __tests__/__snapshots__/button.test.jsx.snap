// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button (Snapshot) Button renders a native button with given props 1`] = `
<button
  className={undefined}
  name="testButton"
  onClick={undefined}
/>
`;
